apps/studio/src/features/editing-experience/components/ComplexEditorStateDrawer.tsx

```diff
@@ -61,6 +61,8 @@ export default function ComplexEditorStateDrawer(): JSX.Element {
 
   const { mutateAsync: uploadAsset, isLoading: isUploadingAsset } =
     useUploadAssetMutation({ siteId })
+  const { mutateAsync: deleteAsset, isLoading: isDeletingAsset } =
```

Comment:
~~should we use `mutate` over `mutateAsync`? like this if anything throws it will crash the app since errors will not be handled by react-query.~~

~~See [Mastering Mutations in React Query section](https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutate-or-mutateasync)~~

edit: nvm, saw that you are firing off dependent mutations. async seems to be the easiest way for that.